docs: add community health files, templates, and PR enforcement workflow

sunydepalpur opened the pull request:
# PR Description: Project Governance and CI/CD Enhancements

## Summary
This Pull Request introduces comprehensive project governance documents, standardized issue/PR templates, and enhanced CI/CD workflows to ensure code quality, security, and a clean commit history.

## Changes

### ⚖️ Project Governance
- **CODE_OF_CONDUCT.md**: Established community standards and reporting guidelines.
- **SECURITY.md**: Defined the project's security policy and vulnerability reporting process.
- **Templates**: Added standardized templates for Bug Reports, Feature Requests, and Pull Requests.

### 🤖 CI/CD Workflows
- **pr-branch-check.yml**: Refactored to only enforce branch rules for Pull Requests originating from forks.
- **pr-check.yml**:
    - Added **Signed Commit Verification**: Ensures all commits in a PR are cryptographically signed.
    - Added **Semantic PR Title Check**: Enforces Conventional Commits for PR titles to support clean squashing.
    - Expanded triggers to include `staging` and `dev` branches.

### 📝 Documentation
- **CONTRIBUTING.md**:
    - Enforced **Squash and Merge** as the strict merge strategy.
    - Emphasized the requirement for semantic PR titles and signed commits.
- **README.md**: General updates to reflect the new project structure and policies.

## Verification Results
- Verified GitHub Action syntax for all new and modified workflows.
- Confirmed that governance documents are correctly linked and formatted.
- Manually tested GPG signing configuration locally.

## Checklist
- [x] Signed commits enforced
- [x] Semantic PR titles enforced
- [x] Governance documents added
- [x] Branch enforcement restricted to forks

GunniBusch commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,29 @@
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
```
can we have this auto detectable based on style quides and apply label. I really like label and think they are often underused.

pranshuparmar replied:
@sunydepalpur Idea sounds good but I'm sure it'll take a bit of effort. I'm perfectly comfortable with your implementation as well.

sunydepalpur commented on .github/workflows/pr-check.yml:
```diff
@@ -2,13 +2,54 @@ name: PR Check
+            echo "::error::Pull Requests from forks must be opened against the 'staging' branch, not 'main'."
+            exit 1
+          fi
+
```
Don't want to be dependent on external flows, let contributer be aware of the requiremnts

sunydepalpur replied:
fine I will look into it detail, try to steal some

sunydepalpur replied:
I was aware but don't have access to the repo so can't do much, leaving this for you @pranshuparmar, I will also try to get inspiratiosn from homebrew and add more workflows and forms.

GunniBusch commented on .github/workflows/pr-check.yml:
```diff
@@ -2,13 +2,54 @@ name: PR Check
+    steps:
+      - name: Check for signed commits
+        uses: 1Password/check-signed-commits-action@v1
+        
```
Ok but my point is that the requirement for all external contributors to sign their commits is a bit bold move, especialy seeing that most come from TikTok or yt and probably dont even know what signing your commit means.

pranshuparmar replied:
I would side with @GunniBusch here, its better if we don't enforce signed commits as of now. The project is in early stage.

GunniBusch commented on .github/ISSUE_TEMPLATE/bug_report.md:
what's your opinion of using issue forms? https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

sunydepalpur commented:
Thanks for the feedback! @pranshuparmar  🙌 I totally agree—I got a bit carried away, and breaking this up will make it much easier to review and manage.

I’m ready to split this into as many PRs as needed. I’ll keep an eye out for your detailed thoughts later today, and then I’ll draft a plan to migrate these changes over piece by piece.

Appreciate you helping guide this!

pranshuparmar commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,20 @@
+**Describe the solution you'd like**
+A clear and concise description of what you want to happen.
+
+**Describe alternatives you've considered**
```
Might deviate users a bit, better to remove IMHO.

pranshuparmar commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,29 @@
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+## How Has This Been Tested?
```
I would remove this for now, don't want to scare away contributors.

pranshuparmar commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,29 @@
+- [ ] Test B
+
+## Checklist:
+- [ ] My code follows the style guidelines of this project
```
Point for running `go fmt` can be made explicitly either inlined to this or as new.

pranshuparmar commented on CONTRIBUTING.md:
```diff
@@ -32,15 +39,135 @@ go build -ldflags "-X main.version=v0.0.0-dev -X main.commit=$(git rev-parse --s
+> ### Legal Notice
+> When contributing to this project, you must agree that you have authored 100% of the content, that you have the necessary rights to the content and that the content you contribute may be provided under the project license.
+
+### Reporting Bugs
```
I would personally remove entire reporting bugs and enhancements section from `CONTRIBUTING.md`. I completely understand that you're trying to set a proper standard here but I think the templates should be good enough for now.

pranshuparmar commented on CONTRIBUTING.md:
This part can be removed-
```
The PR title will become the final commit message, so ensure it follows our [Commit Messages](#commit-messages) guide.
```

sunydepalpur replied:
updated here - https://github.com/pranshuparmar/witr/pull/111/files

pranshuparmar commented on CONTRIBUTING.md:
Let's remove this for now, we'll see how much its actually being used few months down the line.

pranshuparmar commented on README.md:
```diff
@@ -544,8 +557,22 @@ witr is successful if:
 ---
 
-## 11. AI Assistance Disclaimer
+## 11. Security Policy
```
Similar to license, can be removed from here.

pranshuparmar commented:
@sunydepalpur Apologies for the delayed response and for the number of comments. I wanted to take a proper, unhurried look at this, which is why I waited for the weekend.

Thanks again for the effort you’ve put into this PR, it’s genuinely appreciated.

I think this would be better to merge if we split it into three smaller PRs:
- One for issue and PR templates
- One for the workflow
- One for the remaining documentation files

I’ve left quite a few comments. Please feel free to let me know if you’d prefer that I take some of these up myself. I’m completely fine with you handling none, one, or all three of these parts. Thanks again 🙌.

sunydepalpur commented on README.md:
```diff
@@ -1,5 +1,17 @@
 # witr (why-is-this-running)
 
+
+[![Go Report Card](https://goreportcard.com/badge/github.com/pranshuparmar/witr)](https://goreportcard.com/report/github.com/pranshuparmar/witr)
```
keeping it thanks

sunydepalpur commented:
Thanks @pranshuparmar for the detailed review! I agree that splitting this up makes it much easier to manage.

I've broken the changes down into **5 separate PRs** to keep things focused:

1.  **Security Policy**: Added `SECURITY.md` to define supported versions and reporting process. - https://github.com/pranshuparmar/witr/pull/109
2.  **Documentation**: Synced `README.md` and removed the duplicate security section. - https://github.com/pranshuparmar/witr/pull/110
3.  **Contributing Guide**: Refined [CONTRIBUTING.md](cci:7://file:///Users/someshchoudhary/Documents/Projects/witr/CONTRIBUTING.md:0:0-0:0) to be specific to this project (Go/CLI) and removed generic placeholders. - https://github.com/pranshuparmar/witr/pull/111
4.  **Templates**: Added [bug_report.md](cci:7://file:///Users/someshchoudhary/Documents/Projects/witr/.github/ISSUE_TEMPLATE/bug_report.md:0:0-0:0), [feature_request.md](cci:7://file:///Users/someshchoudhary/Documents/Projects/witr/.github/ISSUE_TEMPLATE/feature_request.md:0:0-0:0), and [PULL_REQUEST_TEMPLATE.md](cci:7://file:///Users/someshchoudhary/Documents/Projects/witr/.github/PULL_REQUEST_TEMPLATE.md:0:0-0:0) to streamline issues and PRs. - https://github.com/pranshuparmar/witr/pull/112
5.  **Workflows**: Added `release.yml` and updated [pr-check.yml](cci:7://file:///Users/someshchoudhary/Documents/Projects/witr/.github/workflows/pr-check.yml:0:0-0:0) (including fixes for `vendor/` formatting checks and making signed commits optional). - https://github.com/pranshuparmar/witr/pull/113

I'll open these shortly and link them here. Thanks again for the guidance! 🙌
